pkg/controllers/dynakube/otelc/statefulset/env.go

```diff
@@ -107,3 +120,33 @@ func getEnvs(dk *dynakube.DynaKube) []corev1.EnvVar {
 
 	return envs
 }
+
+func getDynakubeProxyEnvValue(envVar string, src *value.Source) corev1.EnvVar {
+	if src.ValueFrom != "" {
```

Comment:
since it's Pointer it can panic, better to check if it's not nill first and the do src.ValueFrom.

Reply:
I thought about this and deliberately decided against. This function is only called if a proxy is configured. The check is done by the `dk.HasProxy()` call above. The only options to report an error from this function would be an error return parameter or returning a pointer that can be nil. All of this defies the purpose of this function - convenience and reduce code duplication.